src/test/java/org/springframework/data/mapping/PreferredConstructorDiscovererUnitTests.java

```diff
@@ -149,6 +150,34 @@ void detectsMetaAnnotatedValueAnnotation() {
 		});
 	}
 
+	@Test // GH-2332
```

Comment:
I'd like to see a test for the behavior of a private all args constructor for Records, because right now I can't easily tell how the code behaves.

Reply:
Constructors that match the canonical constructor cannot be private. Also, a constructor matching the canonical makes Java backoff from its default constructor creation (same as with public no-arg constructors for regular classes).